asynch/proto/streams/buffered.py

```diff
@@ -145,8 +146,8 @@ async def read_varint(self):
 
         if packets:
             return leb128.u.decode(packets)
-        else:
-            return None
+        msg = "cannot read an integer from empty packets"
```

Comment:
```suggestion
        msg = "Failed to read data from socket. Connection closed?"
```

Reply:
Why guessing if reading an integer from empty packets considered an error. I am not sure about other possible outcomes resulting in the same empty packets case.

Anyway, please do as you like since it is your work and your decisions take precedence.